src/frontend/sharedComponents/CustomHeaderLeftClose.tsx

```diff
@@ -243,19 +243,19 @@ const SharedBackButton = ({
 function checkEqual(
   original: Observation,
   {
-    editted,
-    numberOfPhotos,
+    edited,
+    numberOfAttachments,
   }: {
-    editted: Observation | ClientGeneratedObservation | null;
-    numberOfPhotos?: number;
+    edited: Observation | ClientGeneratedObservation | null;
+    numberOfAttachments?: number;
   },
 ) {
-  if (!editted || !('docId' in editted)) return true;
-  // attachments are created right before an observation is made, so we need to check # photos that are about to be saved
-  const {attachments: originalAtts, ...orignalNoPhotos} = original;
+  if (!edited || !('docId' in edited)) return true;
+  // attachments are created right before an observation is made, so we need to check # attachments that are about to be saved
+  const {attachments: originalAtts, ...originalNoAttachments} = original;
 
-  if (originalAtts.length !== numberOfPhotos) return false;
+  if (originalAtts.length !== numberOfAttachments) return false;
 
-  const {attachments, ...edittedNoPhotos} = editted;
-  return isEqual(orignalNoPhotos, edittedNoPhotos);
+  const {attachments, ...editedNoAttachments} = edited;
+  return isEqual(originalNoAttachments, editedNoAttachments);
```

Comment:
Sorry, this logic doesnt work anymore. originally you could not delete photos, so i was just checking the # of photos. If a user added a photo, the original number of photos would be different from the editted number of photos.

But now a user can delete an audio file. So in theory a user could create an new audio file and delete an old audio file. That would mean the # of attachments would be the same so `isEqual` would return to true even though it is false. So we could actually just use `isEqual` on the `original`  and `edittted`. I didnt just do that originally because I was worried that the `attachments` array order would effect `isEqual` but its better for the user to get a false negative then a false positive.

Reply:
I made the fix you suggested and tested it out and it seems to work.